src/components/ModelSelector.jsx

```diff
@@ -7,31 +7,37 @@ import {
   ToggleButton,
 } from "react-bootstrap";
 
-import { useSelectorContext } from "../context/selector-context";
-import { colorMaps, measurement, season, tide } from "../assets/config.json";
-import { useControlsContext } from "../context/controls-context";
-
-export default function ModelSelector({ toggleControls }) {
+export default function ModelSelector(props) {
   const {
-    state: { selection },
-    dispatch: selectorDispatch,
-  } = useSelectorContext();
+    BUTTONS,
+    ALL_COLOR_MAPS,
+    selection,
+    setSelection,
+    setColorMap,
+    toggleControls,
+  } = props;
 
-  const { dispatch: controlsDispatch } = useControlsContext();
+  function handleMeasurementChange(val) {
+    setSelection({
+      ...selection,
+      measurement: val,
+    });
```

Comment:
should this be a function that takes `selection` as an arg?

Reply:
I was missing `selection` but it's a callback not an argument to `setSelection`. `React.useState()` updates state variables asynchronously - adding the callback in `setSelection` forces React to finish any updates before executing the function

```js
setSelection((selection) => ({
      ...selection,
      [button]: value,
}));
```